flink-core/src/main/java/org/apache/flink/api/common/io/FileInputFormat.java

```diff
@@ -461,8 +463,9 @@ public LocatableInputSplitAssigner getInputSplitAssigner(FileInputSplit[] splits
 		
 		// take the desired number of splits into account
 		minNumSplits = Math.max(minNumSplits, this.numSplits);
-		
-		final Path path = this.filePath;
+
+		final Path path = new Path(URLDecoder.decode(this.filePath.toString(), Charset.defaultCharset().name()));
```

Comment:
Typically, when going from flink paths to Strings to jave file API's on Windows you get exceptions since Flink paths on Windows always start with a "/". But this call never goes through the file API, hence it does succeed.

Anyway, an easier solution is to simply not call `URI#toString` when calling the `Path` constructor in the tests. Then the spaces aren't escaped and the tests succeed.